lib/vdc/UnstructuredGrid3D.cpp

```diff
@@ -0,0 +1,278 @@
+  _zug(zug)
+{
+    VAssert(xug.GetDimensions().size() == 1);
+    VAssert(yug.GetDimensions().size() == 1);
+    VAssert(zug.GetDimensions().size() == 1);
+
+    VAssert(location == NODE);
+}
+
+
+vector<size_t> UnstructuredGrid3D::GetCoordDimensions(size_t dim) const
+{
+    if (dim == 0) {
+        return _xug.GetDimensions();
+    } else if (dim == 1) {
+        return _yug.GetDimensions();
+    } else if (dim == 2) {
+        return _zug.GetDimensions();
+    } else {
+        return (vector<size_t>(1, 1));
+    }
+}
+
+
+size_t UnstructuredGrid3D::GetGeometryDim() const { return (3); }
+
+
+void UnstructuredGrid3D::GetUserExtentsHelper(DblArr3 &minu, DblArr3 &maxu) const
+{
+    float range[2];
+
+    _xug.GetRange(range);
+    minu[0] = range[0];
+    maxu[0] = range[1];
+
+    _yug.GetRange(range);
+    minu[1] = range[0];
+    maxu[1] = range[1];
+
+    _zug.GetRange(range);
+    minu[2] = range[0];
+    maxu[2] = range[1];
+}
+
+
+void UnstructuredGrid3D::GetBoundingBox(const Size_tArr3 &min, const Size_tArr3 &max, DblArr3 &minu, DblArr3 &maxu) const
+{
+    Size_tArr3 cMin;
+    ClampIndex(min, cMin);
+
+    Size_tArr3 cMax;
+    ClampIndex(max, cMax);
+
+    float range[2];
+
+    _xug.GetRange(min, max, range);
+    minu[0] = range[0];
+    maxu[0] = range[1];
+
+    _yug.GetRange(min, max, range);
+    minu[1] = range[0];
+    maxu[1] = range[1];
+
+    _zug.GetRange(min, max, range);
+    minu[2] = range[0];
+    maxu[2] = range[1];
+
+    //    printf("UnstructuredGrid3D::%s(-, -)\n", __func__);
+    //    printf("\t min = [%f, %f, %f]\n", minu[0], minu[1], minu[2]);
+    //    printf("\t max = [%f, %f, %f]\n", maxu[0], maxu[1], maxu[2]);
+}
+
+
+bool UnstructuredGrid3D::GetEnclosingRegion(const DblArr3 &minu, const DblArr3 &maxu, Size_tArr3 &min, Size_tArr3 &max) const
+{
+    VAssert(0 && "Not implemented");
+    return (true);
+}
+
+
+void UnstructuredGrid3D::GetUserCoordinates(const Size_tArr3 &indices, DblArr3 &coords) const
+{
+    Size_tArr3 cIndices;
+    ClampIndex(indices, cIndices);
+
+    coords[0] = _xug.GetValueAtIndex(cIndices);
+    coords[1] = _yug.GetValueAtIndex(cIndices);
+    coords[2] = _zug.GetValueAtIndex(cIndices);
+}
+
+
+bool UnstructuredGrid3D::GetIndicesCell(const DblArr3 &coords, Size_tArr3 &indices) const
+{
+    vector<size_t> nodes2D;
+    vector<double> lambda;
+    float          zwgt[2];
+
+    return (_insideGrid(coords, indices, nodes2D, lambda, zwgt));
+}
+
+
+bool UnstructuredGrid3D::InsideGrid(const DblArr3 &coords) const
+{
+    Size_tArr3          indices;
+    std::vector<size_t> nodes2D;
+    vector<double>      lambda;
+    float               zwgt[2];
+
+    return (_insideGrid(coords, indices, nodes2D, lambda, zwgt));
+}
+
+
+bool UnstructuredGrid3D::_insideGrid(const DblArr3 &coords, Size_tArr3 &cindices, std::vector<size_t> &nodes2D, std::vector<double> &lambda, float zwgt[2]) const
+{
+    VAssert(_location == NODE);
+    VAssert(!"Not implemented");
```

Comment:
How about deleting the rest of the body of this function since it's not being used? If it's a WIP than maybe just ifdef it out.

Reply:
Done